File: app/src/main/java/com/example/website/consulta/Emision.java
package com.example.website.consulta;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.view.Window;
import android.widget.Button;

public class Emision extends AppCompatActivity {

    Button nuevo,consulta,anular,modificar;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_emision);

        nuevo     = findViewById(R.id.Nueva_v);
        consulta  = findViewById(R.id.Consultar_v);
        anular    = findViewById(R.id.Anular_V);
        modificar = findViewById(R.id.Modificar_v);

        nuevo.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent n = new Intent(Emision.this,Nventas.class);
                startActivity(n);
            }
        });

        consulta.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent c = new Intent(Emision.this,Consultarve.class);
                startActivity(c);
            }
        });
        anular.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent a = new Intent(Emision.this,Anularve.class);
                startActivity(a);
            }
        });
        modificar.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent m = new Intent(Emision.this,Modificarve.class);
                startActivity(m);
            }
        });
    }
}

File: app/src/main/java/com/example/website/consulta/View/Emision.kt
package com.example.website.consulta.View

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.GridLayoutManager
import com.example.website.consulta.Model.Entidad.TIPO_DOCUMENTO
import com.example.website.consulta.Model.Entidad.TIPO_PROCESO_VENTA
import com.example.website.consulta.Model.Entidad.VentasProceso
import com.example.website.consulta.Model.VentasProcesoProvider
import com.example.website.consulta.View.Adapter.VentasProcesoAdapter
import com.example.website.consulta.databinding.ActivityEmisionBinding

class Emision : AppCompatActivity() {
    private lateinit var binding: ActivityEmisionBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityEmisionBinding.inflate(layoutInflater)
        setContentView(binding.root)
        initRecycleView()
    }

    private fun initRecycleView() {
        binding.recycleVentasProces.layoutManager = GridLayoutManager(this, 2)
        binding.recycleVentasProces.adapter =
            VentasProcesoAdapter(VentasProcesoProvider.lstProcesosVentas) { ventasProceso ->
                onItemSelected(
                    ventasProceso
                )
            }
    }

    private fun onItemSelected(ventasProceso: VentasProceso) {
        if (ventasProceso.tipoProcesoVenta is TIPO_DOCUMENTO) {
            val c = Intent(this@Emision, Nventas::class.java)
            c.putExtra("TipoDocumento", ventasProceso.tipoProcesoVenta)
            startActivity(c)
        } else if (ventasProceso.tipoProcesoVenta is TIPO_PROCESO_VENTA) {
            val c = Intent(this@Emision, CancelacionDocumentos::class.java)
            startActivity(c)
        }
    }
}